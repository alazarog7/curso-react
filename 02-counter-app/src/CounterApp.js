import React, { useState } from 'react'; // {} -> es para importar algo en especifico
import PropType from 'prop-types';

const CounterApp = ({value})=> {

    const [counter, setCounter] = useState(value);

    const inc = () => {
        setCounter(counter+1)
    }

    const dec = () => {
        setCounter(counter-1)
    }

    const res = () => {
        setCounter(value - value)
    }

    return (
        <>
            <h1> Counter App </h1>
            <h2>{counter}</h2>
            <button onClick={inc}>+1</button>
            <button onClick={res}>Reset</button>
            <button onClick={dec}>-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropType.number
}

CounterApp.defaultProps = {
    value: 0
}
export default CounterApp;