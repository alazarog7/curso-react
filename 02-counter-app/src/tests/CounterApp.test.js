import { shallow } from "enzyme"
import CounterApp from "../CounterApp"

describe('Pruebas del CounterApp',() => {

    let wrapper = shallow(<CounterApp/>);

    beforeEach(()=>{
        wrapper = shallow(<CounterApp/>);
    })

    afterEach(()=>{
        wrapper = null;
    })

    test('debe mostrar la plantilla del contandor ', () => {
       
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar el valor de 100 en el contador', ()=>{
        wrapper = shallow(<CounterApp value={100}/>);
        const numeroInicial = wrapper.find('h2').text().trim();
        expect(numeroInicial).toBe("100");
    });
    
    test('debe incrementar con el boton +1',()=>{
        wrapper.find('button').at(0).simulate('click');
        const counter = wrapper.find('h2').text().trim();
        expect(counter).toEqual("1");

    })

    test('debe incrementar con el boton -1',()=>{
        wrapper.find('button').at(2).simulate('click');
        const counter = wrapper.find('h2').text().trim();
        expect(counter).toEqual("-1");
    });

    test('debe incrementar con el boton -1',()=>{
        wrapper.find('button').at(1).simulate('click');
        const counter = wrapper.find('h2').text().trim();
        expect(counter).toEqual("0");
    });
})