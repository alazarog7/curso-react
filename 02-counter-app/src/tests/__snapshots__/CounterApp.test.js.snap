// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del CounterApp debe mostrar la plantilla del contandor  1`] = `
<Fragment>
  <h1>
     Counter App 
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
